elements/summonindicator.lua

```diff
@@ -0,0 +1,92 @@
+--[[
+# Element: SummonIndicator
+
+TBD
+
+--]]
+
+local _, ns = ...
+local oUF = ns.oUF
+
+-- sourced from Blizzard_APIDocumentation/IncomingSummonDocumentation.lua
+local SUMMON_STATUS_NONE = Enum.SummonStatus.None or 0
+local SUMMON_STATUS_PENDING = Enum.SummonStatus.Pending or 1
+local SUMMON_STATUS_ACCEPTED = Enum.SummonStatus.Accepted or 2
+local SUMMON_STATUS_DECLINED = Enum.SummonStatus.Declined or 3
+
+local function Update(self)
+	local element = self.SummonIndicator
+
+	--[[ Callback: SummonIndicator:PreUpdate()
+	Called before the element has been updated.
+
+	* self - the SummonIndicator element
+	--]]
+	if(element.PreUpdate) then
+		element:PreUpdate()
+	end
+
+	local status = C_IncomingSummon.IncomingSummonStatus(self.unit)
+	if(status ~= SUMMON_STATUS_NONE) then
+		if(status == SUMMON_STATUS_PENDING) then
+			element:SetAtlas('Raid-Icon-SummonPending')
+		elseif(status == SUMMON_STATUS_ACCEPTED) then
+			element:SetAtlas('Raid-Icon-SummonAccepted')
+		elseif(status == SUMMON_STATUS_DECLINED) then
+			element:SetAtlas('Raid-Icon-SummonDeclined')
+		end
+
+		element:Show()
+	else
+		element:Hide()
+	end
+
+	--[[ Callback: SummonIndicator:PostUpdate(status)
+	Called after the element has been updated.
+
+	* self  - the SummonIndicator element
+	* status - the unit's incoming summon status (number)[0-3]
+	--]]
+	if(element.PostUpdate) then
+		return element:PostUpdate(status)
+	end
+end
+
+local function Path(self, ...)
+	--[[ Override: SummonIndicator.Override(self, event)
+	Used to completely override the internal update function.
+
+	* self  - the parent object
+	* event - the event triggering the update (string)
+	* ...   - the arguments accompanying the event
+	--]]
+	return (self.SummonIndicator.Override or Update) (self, ...)
+end
+
+local function ForceUpdate(element)
+	if(not element.__owner.unit) then return end
+	return Path(element.__owner, 'ForceUpdate')
+end
+
+local function Enable(self)
+	local element = self.SummonIndicator
+	if(element) then
+		element.__owner = self
+		element.ForceUpdate = ForceUpdate
+
+		self:RegisterEvent('INCOMING_SUMMON_CHANGED', Path, true)
```

Comment:
ISC is not a unitless event according to the docs: https://github.com/tomrus88/BlizzardInterfaceCode/blob/master/Interface/AddOns/Blizzard_APIDocumentation/UnitDocumentation.lua#L304

However it won't work, because our event code is smarter than us: https://github.com/oUF-wow/oUF/blob/master/events.lua#L80-L83 (or add the event to https://github.com/oUF-wow/oUF/blob/master/events.lua#L15 but I think this is easy to get out of sync). Btw, the last time I checked, you could register unitless events with units just fine (frame:RegisterUnitEvent('SPELLS_CHANGED', 'player') worked), so maybe tell our event code to not take decisions on the user's account.